### PR TITLE
NADAM Optimizer

### PR DESCRIPTION
**Pull request by just-a-jazz**
This adds the feature requested by #7715

### CODE REVIEW

**Comment by alextp** [comment]
Thanks for the contribution!

I think adding more booleans flags to the adam code is a little sad. How do you feel about making this a separate optimizer (sharing the op kernel is ok) in tensorflow/contrib/opt?

**Comment by alisidd** [comment]
@alextp I can definitely do that

File: tensorflow/python/training/adam.py [hunk 1]
```diff
@@ -36,7 +36,7 @@ class AdamOptimizer(optimizer.Optimizer):
   """
 
   def __init__(self, learning_rate=0.001, beta1=0.9, beta2=0.999, epsilon=1e-8,
-               use_locking=False, name="Adam"):
+               use_locking=False, use_nesterov=False, name="Adam"):
```
Comment (alextp) [inline tensorflow/python/training/adam.py]: Undo the change here as well?

**Comment by alisidd** [comment]
@alextp Is this what you were looking for?

File: tensorflow/python/training/adam_test.py [hunk 2]
```diff
@@ -40,19 +40,25 @@ def adam_update_numpy(param,
                       alpha=0.001,
                       beta1=0.9,
                       beta2=0.999,
-                      epsilon=1e-8):
+                      epsilon=1e-8,
+                      use_nesterov=False):
```
Comment (alextp) [inline tensorflow/python/training/adam_test.py]: Undo the changes here?

File: tensorflow/contrib/opt/python/training/nadam_optimizer_test.py [hunk 3]
```diff
@@ -0,0 +1,298 @@
...
+            update2.run()
+
+          var0_np, m0, v0 = nadam_update_numpy(var0_np, grads0_np, t, m0, v0)
+          var1_np, m1, v1 = nadam_update_numpy(var1_np, grads1_np, t, m1, v1)
+
+          # Validate updated params
+          self.assertAllCloseAccordingToType(var0_np, var0.eval())
+          self.assertAllCloseAccordingToType(var1_np, var1.eval())
+
+  def testTwoSessions(self):
```
Comment (alextp) [inline tensorflow/contrib/opt/python/training/nadam_optimizer_test.py]: Remove this test?

**Review by alextp — COMMENTED** [review]
Looking good. I'd just undo the changes in adam.py and the test, and remove some maybe unnecessary tests from the nadam_test.

**Review by alextp — APPROVED** [review]

**Comment by alextp** [comment]
Jenkins, test this please.

**Comment by alisidd** [comment]
@alextp Oh I just saw your message here, my bad

**Comment by alextp** [comment]
Oh, sorry, I think having nadam_test is good, with a basic test of the math (but no need to repeat all tests from adam_test since there are many)

**Comment by alisidd** [comment]
Probably why the checks are failing haha, which tests should I keep in the tester?

**Comment by alextp** [comment]
testSparse and testBasic

**Comment by alextp** [comment]
Jenkins, test this please.

**Comment by alextp** [comment]
Jenkins, test this please.

**Comment by alisidd** [comment]
Jenkins, test this please.

**Comment by alextp** [comment]
Jenkins, test this please.

**Comment by alextp** [comment]
Jenkins, test this please.

**Comment by alextp** [comment]
Can you fix the error in https://ci.tensorflow.org/job/tensorflow-pull-requests-sanity/4287/console ?

**Comment by alextp** [comment]
Jenkins, test this please.
